contracts/test/integration/IntegrationTestVoltV2.t.sol

```diff
@@ -0,0 +1,375 @@
+        volt.mint(address(this), 1e18);
+
+        vm.expectRevert("Volt: burn amount exceeds balance");
+        volt.burn(2e18);
+    }
+
+    function testBurnFailZeroAddress() public {
+        vm.startPrank(address(0));
+        vm.expectRevert("Volt: cannot burn from the zero address");
+        volt.burn(1e18);
+        vm.stopPrank();
+    }
+
+    function testBurnFrom(uint256 voltToBurn) public {
+        vm.assume(voltToBurn < type(uint256).max); // to make sure we don't run into infinite approval counter example
+        address from = address(0xFFF);
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(from, voltToBurn);
+
+        vm.prank(from);
+        volt.approve(address(this), voltToBurn);
+        assertEq(volt.allowance(from, address(this)), voltToBurn);
+
+        volt.burnFrom(from, voltToBurn);
+
+        assertEq(volt.balanceOf(from), 0);
+        assertEq(volt.allowance(from, address(this)), 0);
+        assertEq(volt.totalSupply(), 0);
+    }
+
+    function testBurnFromFailAddressZero() public {
+        address from = address(0);
+
+        vm.prank(from);
+        volt.approve(address(this), 1e18);
+        assertEq(volt.allowance(from, address(this)), 1e18);
+
+        vm.expectRevert("Volt: cannot burn from the zero address");
+        volt.burnFrom(from, 1e18);
+    }
+
+    function testBurnFromInfiniteApproval(uint256 voltToBurn) public {
+        address from = address(0xFFF);
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(from, voltToBurn);
+
+        vm.prank(from);
+        volt.approve(address(this), type(uint256).max);
+        assertEq(volt.allowance(from, address(this)), type(uint256).max);
+
+        volt.burnFrom(from, voltToBurn);
+
+        assertEq(volt.balanceOf(from), 0);
+        assertEq(volt.allowance(from, address(this)), type(uint256).max);
+        assertEq(volt.totalSupply(), 0);
+    }
+
+    function testBurnFromFailInsufficientBalance() public {
+        address from = address(0xFFF);
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(from, 1e18);
+
+        vm.prank(from);
+        volt.approve(address(this), 2e18);
+
+        vm.expectRevert("Volt: burn amount exceeds balance");
+        volt.burnFrom(from, 2e18);
+    }
+
+    function testBurnFromFailInsufficientAllowance() public {
+        address from = address(0xFFF);
+
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(from, 1e18);
+
+        vm.prank(from);
+        volt.approve(address(this), 0.9e18);
+
+        vm.expectRevert(stdError.arithmeticError);
+        volt.burnFrom(from, 1e18);
+    }
+
+    function testApprove(uint256 voltToApprove) public {
+        assertTrue(volt.approve(address(0xFFF), voltToApprove));
+
+        assertEq(volt.allowance(address(this), address(0xFFF)), voltToApprove);
+    }
+
+    function testTransfer(uint256 voltToTransfer) public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(address(this), voltToTransfer);
```

Comment:
verify the user has been properly check pointed in the voting system. that should be verified along with the total supply incrementing.